injectproxy/routes.go

```diff
@@ -198,17 +198,46 @@ func NewRoutes(upstream *url.URL, label string, opts ...Option) (*routes, error)
 
 func (r *routes) enforceLabel(h http.HandlerFunc) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		queryLabel := r.label
 		lvalue := req.FormValue(r.label)
+		matcherType := labels.MatchEqual
+		if strings.HasPrefix(lvalue, "~") {
+			matcherType = labels.MatchRegexp
+			lvalue = strings.TrimPrefix(lvalue, "~")
+		}
+		if lvalue == "" {
+			queryLabel = r.label + "!"
+			lvalue = req.FormValue(queryLabel)
+			matcherType = labels.MatchNotEqual
+			if lvalue == "" {
+				queryLabel = r.label + "!~"
+				for k, v := range req.Form {
+					if strings.HasPrefix(k, queryLabel) {
+						if v[0] == "" {
```

Comment:
I think it should be 

```suggestion
						if v[0] != "" {
```